velox/exec/tests/utils/DistributedPlanBuilder.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/tests/utils/DistributedPlanBuilder.h"
+
+namespace facebook::velox::exec::test {
+
+DistributedPlanBuilder::DistributedPlanBuilder(
+    const runner::MultiFragmentPlan::Options& options,
+    std::shared_ptr<core::PlanNodeIdGenerator> planNodeIdGenerator,
+    memory::MemoryPool* pool)
+    : PlanBuilder(planNodeIdGenerator, pool), options_(options) {
+  auto* root = rootBuilder();
+  VELOX_CHECK(
+      root->stack_.empty(),
+      "Cannot make a top level DistributedPlanBuilder inside an open one");
+  root->stack_.push_back(this);
+  newFragment();
+  current_.width = options_.numWorkers;
+}
+
+DistributedPlanBuilder::DistributedPlanBuilder(DistributedPlanBuilder& parent)
+    : PlanBuilder(parent.planNodeIdGenerator(), parent.pool()),
+      options_(parent.options_),
+      parent_(&parent) {
+  auto* root = rootBuilder();
+  root->stack_.push_back(this);
+  newFragment();
+  current_.width = options_.numWorkers;
+}
+
+std::vector<runner::ExecutableFragment> DistributedPlanBuilder::fragments() {
+  newFragment();
+  return std::move(fragments_);
+}
+
+void DistributedPlanBuilder::newFragment() {
+  if (!current_.taskPrefix.empty()) {
```

Comment:
taskPrefix is always set for a executable fragment? Can we make a pointer to explicitly check on that?